### PR TITLE
[#896] Added table with additional fields to case status-page dashboard

### PR DESCRIPTION
**Pull request by Bartvaderkin**

### CODE REVIEW

**Comment by Bartvaderkin** [comment]
This needs a little bit of styling from @svenvandescheur, it can be added to this PR or done after merge.

File: src/open_inwoner/components/templatetags/dashboard_tags.py [hunk 1]
```diff
@@ -59,3 +68,71 @@ def case_dashboard(case: dict, **kwargs) -> dict:
...
+    Variables:
+        + case: dict | The case to be able to build the dashboard, fetching the documents and statusses of the case.
+
+    Extra context:
+        + table: TableConfig | The configuration of the table.
+    """
+
+    # build rows for data we actually have
+    rows = []
+    if case.get("identification"):
```
Comment (annashamray) [inline src/open_inwoner/components/templatetags/dashboard_tags.py:90]:
```suggestion
    if case.get("initiator"):
```

File: src/open_inwoner/openzaak/utils.py [hunk 2]
```diff
@@ -52,6 +53,54 @@ def is_zaak_visible(zaak: Zaak) -> bool:
     return is_object_visible(zaak, config.zaak_max_confidentiality)
 
 
+def get_role_identification_display(rol: Rol) -> str:
```
Comment (annashamray) [inline src/open_inwoner/openzaak/utils.py:56]: this comment is optional, but it looks like this function will be better as `Rol` method

File: src/open_inwoner/openzaak/utils.py [hunk 3]
```diff
@@ -52,6 +53,54 @@ def is_zaak_visible(zaak: Zaak) -> bool:
...
+    elif rol.betrokkene_type == RolTypes.medewerker:
+        display = join(
+            value("voorletters"),
+            value("voorvoegsel_achternaam"),
+            value("achternaam"),
+        )
+
+    if not display:
+        # fallback to generic role description
+        return RolTypes.labels.get(rol.betrokkene_type, rol.betrokkene_type)
```
Comment (annashamray) [inline src/open_inwoner/openzaak/utils.py:99]:
```suggestion
        return rol.get_betrokkene_type_display()
```

File: src/open_inwoner/openzaak/utils.py (hunk 2 above)
Comment (annashamray) [inline src/open_inwoner/openzaak/utils.py:56]:
```suggestion
def get_role_name_display(rol: Rol) -> str:
```
You don't show here any id fields, so I think that `name` or `description` is better here

File: src/open_inwoner/openzaak/tests/test_utils.py [hunk 4]
```diff
@@ -93,3 +97,105 @@ def test_filter_info_object_visibility(self):
                 ),
             )
             self.assertFalse(is_info_object_visible(info_object, "non_existent_key"))
+
+    def test_get_role_identification_display(self):
```
Comment (annashamray) [inline src/open_inwoner/openzaak/tests/test_utils.py:101]: Nice test coverage!

**Review by annashamray — CHANGES_REQUESTED** [review]
I personally think that the styling can be also done in this PR
But it's up to you :)

File: src/open_inwoner/components/templatetags/dashboard_tags.py [hunk 5]
```diff
@@ -16,6 +16,15 @@ class DashboardConfig(TypedDict):
     metrics: list[Metric]
 
 
+class Row(TypedDict):
```
[the remark this answers is not in the document]
Comment (Bartvaderkin) [reply src/open_inwoner/components/templatetags/dashboard_tags.py:19]: This is the custom type hint, it isn't real class but a PEP 589 TypedDict. 

So it doesn't product an instance with methods but just a dictionary. I did it to follow the pattern from the existing code.

File: src/open_inwoner/components/templatetags/dashboard_tags.py [hunk 6]
```diff
@@ -59,3 +68,71 @@ def case_dashboard(case: dict, **kwargs) -> dict:
...
+        )
+    if case.get("end_date_legal"):
+        rows.append(
+            {
+                "label": _("Wettelijke termijn"),
+                "value": case.get("end_date_legal"),
+            }
+        )
+
+    table: TableConfig = {"rows": rows}
```
[the remark this answers is not in the document]
Comment (Bartvaderkin) [reply src/open_inwoner/components/templatetags/dashboard_tags.py]: A TypedDict isn't a normal class and doesn't produce a real instance but a normal dict. But I'll make it dryer another way.

File: src/open_inwoner/openzaak/utils.py (hunk 2 above)
Comment (Bartvaderkin) [reply src/open_inwoner/openzaak/utils.py]: If this works how we want we could move it to the zgw consumers.

**Comment by Bartvaderkin** [comment]
@annashamray I applied your suggestions and replied to your notes on the TypedDict.

File: src/open_inwoner/components/templatetags/dashboard_tags.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (annashamray) [reply src/open_inwoner/components/templatetags/dashboard_tags.py:19]: right, sorry, I didn't notice that it inherits from `TypedDict`

**Review by annashamray — APPROVED** [review]
:+1:

File: src/open_inwoner/components/templatetags/dashboard_tags.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (annashamray) [reply src/open_inwoner/components/templatetags/dashboard_tags.py]: Right, my bad, I haven't paid attention that it inherits from `TypedDict`

**Comment by Bartvaderkin** [comment]
Ok, lets merge so sven can style in develop.

**Pull request merged**